### ISSUE TITLE
npm install fails with node 10

### ISSUE THREAD

**Issue by kmacphee** [angular-in-action/datacenter#2]
Running 'npm install' in this repo seems to fail with node 10:

> datacenter kirk$ npm version
> { datacenter: '0.0.0',
>   npm: '5.6.0',
>   ares: '1.14.0',
>   cldr: '33.0',
>   http_parser: '2.8.0',
>   icu: '61.1',
>   modules: '64',
>   napi: '3',
>   nghttp2: '1.29.0',
>   node: '10.0.0',
>   openssl: '1.1.0h',
>   tz: '2018c',
>   unicode: '10.0',
>   uv: '1.20.2',
>   v8: '6.6.346.24-node.5',
>   zlib: '1.2.11' }

'npm install' output from my machine is in the following Gist: https://gist.github.com/kmacphee/0b368a1155ea3d2ba4bb61e633a5dd43

**Comment by gnomeontherun** [comment]
Reading the error it appears to say that you need to install Xcode CLI tools first, which is just part of what is required for building a lot of Node packages on Mac.

I believe you can run `xcode-select --install` and then try to npm install again.

**Comment by kmacphee** [comment]
Hi Jeremy, thanks for the suggestion. 

I think the message 'xcode-select' threw is pretty benign, it's because I only had the command line tools configured (which works) rather than the full application on my Mac. Anyway, I've now configured the full Xcode preferences to keep 'xcode-select' quiet and updated the Gist.

The real error is: **../../nan/nan_maybe_43_inl.h:112:15: error: no member named 'ForceSet' in 'v8::Object'**

Which I think is related to:

- https://github.com/nodejs/nan/issues/504
- Similar: https://github.com/mapnik/node-mapnik/issues/833

I hope that helps.

**Comment by maisonbleue** [comment]
Hi Kirk did you manage to run through this error ?
I have the same issue and I dont know ho to resolve it.

**Comment by kmacphee** [comment]
Hi Vincent. I've managed to figure out that 'npm install' wants to install 'fsevents' for some reason, but it's not the latest version (which does seem to support Node 10), instead it's version 1.1.2 which uses a deprecated feature/function.

To be fair to Jeremy, he's not directly referencing 'fsevents' in this project. Being quite new to Node though, I'm at a bit of a loss as to what is referencing it and why. There are a lot of packages in the dependency tree, so tracing it will be a bit difficult.

**Comment by maisonbleue** [comment]
Hi again,
As you said It's all about 'fsevents' I managed to install the latest version using 'npm install fsevents@latest'.
Now I dont have the error anymore thank you !

**Comment by gnomeontherun** [comment]
You can debug which package wants to install a dependency by running `npm ls fsevents`, which shows the `@angular/compiler-cli` package uses `chokidar` which uses `fsevents`. I haven't updated these samples yet to v6, which should be largely painless (except some rxjs changes). 

So your solution to force install a newer version might have worked here but doesn't always if the dependency requires a stricter version range to be installed. So I'm going to leave this open as a marker for when I update everything to v6.

**Comment by kmacphee** [comment]
In the end I downgraded back to node v8.11.1 to keep making progress. Lesson learned - I'll stick to the LTS release in future.

I had a little further trouble with `@angular/language-service` not being the same version as many of the other Angular modules - it's "^4.2.4" in package.json whereas everything else is "^5.0.0-beta.6". I upgraded all the Angular bits to v5.2.10 myself and it runs fine now.

How fast Node/Angular development moves has been a bit of a baptism of fire, but I've learned a lot from the struggle!
